lib/internal/http2/compat.js

```diff
@@ -654,13 +656,20 @@ class Http2ServerResponse extends Stream {
       encoding = 'utf8';
     }
 
-    if (this[kState].closed) {
+    if (this[kState].ending) {
+      const err = new ERR_STREAM_WRITE_AFTER_END();
+      if (typeof cb === 'function')
+        process.nextTick(cb, err);
+      this.destroy(err);
+      return false;
+    } else if (this[kState].closed) {
       const err = new ERR_HTTP2_INVALID_STREAM();
       if (typeof cb === 'function')
         process.nextTick(cb, err);
-      else
-        throw err;
-      return;
+      this.destroy(err);
+      return false;
+    } else if (this[kState].destroyed) {
+      return false;
```

Comment:
I think this could look similar to how it did before with `const state = this[kState]` and `let err` and then one block to do the repeated nextTick & destroy stuff. That part seemed cleaner to me.

Reply:
Fixed